packages/babel-helper-member-expression-to-functions/src/index.ts

```diff
@@ -472,14 +493,57 @@ const handle = {
   },
 };
 
+export interface Handler<State> {
+  memoise?(
+    this: HandlerState<State> & State,
+    member: Member,
+    count: number,
+  ): void;
+  destructureSet(
+    this: HandlerState<State> & State,
+    member: Member,
+  ): t.Expression;
+  boundGet(this: HandlerState<State> & State, member: Member): t.Expression;
+  simpleSet?(this: HandlerState<State> & State, member: Member): t.Expression;
+  get(this: HandlerState<State> & State, member: Member): t.Expression;
+  set(
+    this: HandlerState<State> & State,
+    member: Member,
+    value: t.Expression,
+  ): t.Expression;
+  call(
+    this: HandlerState<State> & State,
+    member: Member,
+    args: t.CallExpression["arguments"],
+  ): t.Expression;
+  optionalCall(
+    this: HandlerState<State> & State,
+    member: Member,
+    args: t.OptionalCallExpression["arguments"],
+  ): t.Expression;
+}
+
+export interface HandlerState<State = {}> extends Handler<State> {
+  handle(
+    this: HandlerState<State> & State,
```

Comment:
Q: Are these circular references good for TS?

Reply:
It seems that TS works pretty fine with these circular refs. Are there any better solutions?

Reply:
Not sure there is any🤔